source/server/hot_restarting_parent.cc

```diff
@@ -95,9 +97,12 @@ HotRestartingParent::Internal::getListenSocketsForChild(const HotRestartMessage:
   Network::Address::InstanceConstSharedPtr addr =
       Network::Utility::resolveUrl(request.pass_listen_socket().address());
   for (const auto& listener : server_->listenerManager().listeners()) {
-    if (*listener.get().socket().localAddress() == *addr && listener.get().bindToPort()) {
-      wrapped_reply.mutable_reply()->mutable_pass_listen_socket()->set_fd(
-          listener.get().socket().ioHandle().fd());
+    Network::ListenSocketFactory& socket_factory = listener.get().listenSocketFactory();
+    if (*socket_factory.localAddress() == *addr && listener.get().bindToPort()) {
+      if (socket_factory.sharedSocket().has_value()) {
```

Comment:
Can you add a small comment that we only do socket passing if the listener socket is shared between workers and processes?